extensions/lifecycle/objectProcessor/LifecycleObjectProcessor.js

```diff
@@ -134,8 +136,12 @@ class LifecycleObjectProcessor extends EventEmitter {
         async.parallel([
             done => this._setupConsumers(done),
             done => {
-                this._gcProducer = new GarbageCollectorProducer();
-                this._gcProducer.setupProducer(done);
+                if (this._gcConfig) {
+                    this._gcProducer = new GarbageCollectorProducer();
+                    this._gcProducer.setupProducer(done);
+                } else {
+                    done();
+                }
             },
         ], done);
```

Comment:
I think maybe a cleaner fix for this would be to move the gcProducer creation logic to `LifecycleObjectTransitionProcessor`, this will avoid having to pass arround the gc config everywhere